fsociety.py

```diff
@@ -1311,20 +1380,29 @@ def shellnoob():
 
 
 def androidhash():
-    key = raw_input("Enter the android hash: ")
-    salt = raw_input("Enter the android salt: ")
-    os.system(
-        "git clone --depth=1 https://github.com/PentesterES/AndroidPINCrack.git")
-    os.system(
-        "cd AndroidPINCrack && python AndroidPINCrack.py -H %s -s %s" % (key, salt))
+    key = raw_input("Enter the android hash: ").split(' ')[0]
+    salt = raw_input("Enter the android salt: ").split(' ')[0]
+    symbols = ['!','@','#','$','%','^','&','*','(',')','-','=','+','|','||','&&','/','//','+']
+    if symbols not in key and symbols not in salt:
```

Comment:
The `line` arises a `exception` which prevents the user to make working correctly the tool: 
![Screenshot from 2020-07-26 12-10-37](https://user-images.githubusercontent.com/33063403/88476527-52f38d00-cf39-11ea-8590-42f95e9e97ad.png)
. I would suggest to add something like the following line of code:

```python
key = raw_input("Enter the android hash: ")
salt = raw_input("Enter the android salt: ")
symbols = ['!','@','#','$','%','^','&','*','(',')','-','=','+','|','||','&&','/','//','+', ' '] #Added ' ' to avoid split(' ') which turns `key` and `salt` in a `list`

[symbol for symbol in symbols if symbol not in key and symbol not in salt] == symbols # if True --> the `key` and `salt` strings are correct (don't match any of the `blacklisted` chars)
```

This makes a  `list comprehension` which therefore checks if the string contains one of the characters blacklisted returning the list resulting and then checks if it's equal to the one predefined before ( `symbols` ).